spec.html

```diff
@@ -28,25 +28,19 @@ <h1>String.prototype.replaceAll ( _searchValue_, _replaceValue_ )</h1>
   <p>When the `replaceAll` method is called with arguments _searchValue_ and _replaceValue_, the following steps are taken:</p>
   <emu-alg>
     1. Let _O_ be ? RequireObjectCoercible(*this* value).
-    1. If _searchValue_ is neither *undefined* nor *null*, then
-      1. Let _replacer_ be ? GetMethod(_searchValue_, @@replace).
-      1. If _replacer_ is not *undefined*, then
-        1. Return ? Call(_replacer_, _searchValue_, &laquo; _O_, _replaceValue_ &raquo;).
-    1. Let _string_ be ? ToString(_O_).
-    1. Let _searchString_ be ? ToString(_searchValue_).
-    1. Let _functionalReplace_ be IsCallable(_replaceValue_).
-    1. If _functionalReplace_ is *false*, then
-      1. Set _replaceValue_ to ? ToString(_replaceValue_).
-    1. Search _string_ for the first occurrence of _searchString_ and let _pos_ be the index within _string_ of the first code unit of the matched substring and let _matched_ be _searchString_. If no occurrences of _searchString_ were found, return _string_.
-    1. If _functionalReplace_ is *true*, then
-      1. Let _replValue_ be ? Call(_replaceValue_, *undefined*, &laquo; _matched_, _pos_, _string_ &raquo;).
-      1. Let _replStr_ be ? ToString(_replValue_).
+    1. Let _isRegExp_ be ? IsRegExp(_searchValue_).
+    1. If _isRegExp_ is *true*, then
+      1. Let _flags_ be _searchValue_.[[OriginalFlags]].
+      1. If _flags_ does not contain `"g"`, then
+        1. Set _flags_ to a copy of _flags_.
```

Comment:
Why is this needed? This is a value, not a reference to the internal slot on searchValue.

Alternatively, there’s an existing form of “to the concatenation of a and b” that’s probably better here.

Reply:
> This is a value, not a reference to the internal slot on searchValue.

That’s not what https://tc39.github.io/ecma262/#sec-algorithm-conventions says:

> Algorithm steps may declare named aliases for any value using the form “Let x be someValue”. These aliases are reference-like in that both x and someValue refer to the same underlying data and modifications to either are visible to both. Algorithm steps that want to avoid this reference-like behaviour should explicitly make a copy of the right-hand side: “Let x be a copy of someValue” creates a shallow copy of someValue.

Reply:
Interesting, that seems terrifying :-) either way i think the concatenation language would be simpler (it’s what matchAll uses)